devicetypes/smartthings/aeon-key-fob.src/aeon-key-fob.groovy

```diff
@@ -128,42 +135,51 @@ def configure() {
 
 
 def installed() {
+	state.child_count = 0
 	initialize()
 }
 
 def updated() {
 	initialize()
-	if (!childDevices) {
-		createChildDevices()
-	}
-	else if (device.label != state.oldLabel) {
-		childDevices.each {
-			def segs = it.deviceNetworkId.split("/")
-			def newLabel = "${device.displayName} button ${segs[-1]}"
-			it.setLabel(newLabel)
+	if(state.buttons==4)
+	{
+		if (!state.child_count) {
+			createChildDevices()
+		}
+		else if (device.label != state.oldLabel) {
+			childDevices.each {
+				def segs = it.deviceNetworkId.split("/")
+				def newLabel = "${device.displayName} button ${segs[-1]}"
+				it.setLabel(newLabel)
+			}
+			state.oldLabel = device.label
 		}
-		state.oldLabel = device.label
 	}
 }
 
 def initialize() {
 	// Device only goes OFFLINE when Hub is off
 	sendEvent(name: "DeviceWatch-Enroll", value: JsonOutput.toJson([protocol: "zwave", scheme:"untracked"]), displayed: false)
-
 	def zwMap = getZwaveInfo()
-	def buttons = 4 // Default for Key Fob
-
+	state.oldLabel = device.label
 	// Only one button for Aeon Panic Button
 	if (zwMap && zwMap.mfr == "0086" && zwMap.prod == "0001" && zwMap.model == "0026") {
-		buttons = 1
+		state.buttons = 1
 	}
-	createChildDevices(buttons)
-	sendEvent(name: "numberOfButtons", value: buttons)
+	else
+	{
+		state.buttons = 4
+		if(!state.child_count)
+		{
+			createChildDevices()
+		}
+	}
+	sendEvent(name: "numberOfButtons", value: state.buttons)
 }
 
-private void createChildDevices(def num) {
-	state.oldLabel = device.label
-	for (i in num) {
+private void createChildDevices() {
+	state.child_count = 1
+	for (i in 1..4) {
```

Comment:
As noted before, can we isolate the button count to a model check in one place and make it a variable elsewhere?